text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+   lifecycle so assets could be shared across builds for greater caching
+   benefit. The assets can then be mapped into the build container at the
+   specified location using a volume mount to `pack build`.
+1. Via builder. The builder could come with assets included under
+   `/platform/deps/assets`. This would likely require a builder with many
+   buildpacks to update very frequently though, so may not be the best idea for
+   the average case, however, it could be very useful as a way to distribute
+   dependencies in offline environments.
+1. Via the platform. Platforms may choose to offer enhanced functionality to
+   more easily distribute dependencies. In the end, the platform just needs to
+   ensure that the dependency metadata is available to the buildpack in the
+   required location.
+
+## Implementation
+### Tools
+To start the implementation, we’ll create some tooling to work with the new
+metadata format. We’ll create some basic tools for authors to add metadata,
+update metadata, and package metadata into compressed archives and images. For
+buildpack users, we’ll create some tooling to fetch metadata, overlay metadata
+from multiple sources, pre-cache assets from metadata, and feed metadata and
+pre-cached assets into `pack build`.
+
+### Library Enhancement
+The next step will be to add new functionality in `libpak` and `packit` for
+consuming metadata from the defined metadata location and for consuming
+dependency assets from this metadata. This can be done in an additive way so
+that existing functionality continues to be supported. Support for
+`buildpack.toml` dependency metadata can be deprecated and removed on a
+separate timeline.
+
+### Update Buildpacks
+When we have library support, we’ll be able to update the Paketo buildpacks
+that are installing dependencies to use the metadata provided. For example, we
+can update the Bellsoft Liberica or Amazon Corretteo buildpacks to install
+their JRE/JDK dependencies using this new metadata.
+
+We propose a flag of `BP_EXTERNAL_METADATA_ENABLED` which defaults to `false`
+for use as buildpacks are being converted. In the default state, this flag
+tells a buildpack to use the metadata included with buildpack.toml. When set to
+`true`, a buildpack should use the new metadata. This can provide a way for
+users to test the new functionality without impacting existing users.
+
+### Write Documentation
+Documentation will be added at each step. Initial tool documentation will be on
+the repositories for the tools. The `libpak` and `packit` libraries will update
+their docs as support is added. As individual buildpacks are updated, the
+`README.md` for each buildpack will be updated with the newly supported
+options. When an entire language family is complete, the information will be
+merged into the official Paketo documentation.
+
+### Buildpack Migration Process
+Once a language family has added support for the new metadata format and
```

Comment:
What does it mean for a language family to have support for this new metadata? Language-family buildpacks don't have dependencies, so is the intention here to say: "all buildpacks in a language family that have dependencies have added support for the new mechanism"?

Reply:
This section is to just say that language families have the final say on depreciation. The RFC provides guidance but doesn't mandate anything.

Reply:
I see. Would it be fair to say something like: "once a language-family maintainer group is comfortable that all buildpacks have migrated..."

Reply:
Updated some of the language around here please feel free to take a look!